.github/scripts/setup.sh

```diff
@@ -0,0 +1,260 @@
+#/bin/sh -x
+
+DIR=`dirname $0`
+DISTRO="$1"; shift;
+
+# package management helpers
+PACKAGE_MANAGER_INSTALL_yum="yum --setopt=skip_missing_names_on_install=False install -y"
+PACKAGE_MANAGER_REMOVE_yum="yum erase -y"
```

Comment:
Once we drop CentOS 7 (EOL in a couple of months IIRC) we should replace yum with dnf (it replaced yum several years ago), I don't even have yum on my Fedora 39 boxes.

Reply:
Looks like RedHat rebrands `dnf` to `yum` in RHEL 8: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/managing-software-packages_configuring-basic-system-settings

> Upstream documentation identifies the technology as DNF and the tool is referred to as DNF in the upstream.

Reply:
We can fix this later, I guess. 
As per fedora, in a default rawhide container `yum` is still a symlink to `dnf`, so I don't expect any problems with earlier releases either.
```
:~$ docker run --rm -it fedora:rawhide /usr/bin/yum --version
4.19.0
  Installed: dnf-0:4.19.0-1.fc40.noarch at Mon Feb 19 08:10:37 2024
  Built    : Fedora Project at Thu Feb  8 16:33:42 2024

  Installed: rpm-0:4.19.1.1-1.fc40.x86_64 at Mon Feb 19 08:10:37 2024
  Built    : Fedora Project at Wed Feb  7 15:55:53 2024
```

Reply:
> Looks like RedHat rebrands `dnf` to `yum` in RHEL 8: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/managing-software-packages_configuring-basic-system-settings
> 
> > Upstream documentation identifies the technology as DNF and the tool is referred to as DNF in the upstream.

Well, it's a _version_ of yum that is compatible with dnf (and older yum).

Reply:
> We can fix this later, I guess. As per fedora, in a default rawhide container `yum` is still a symlink to `dnf`, so I don't expect any problems with earlier releases either.
> 
> ```
> :~$ docker run --rm -it fedora:rawhide /usr/bin/yum --version
> 4.19.0
>   Installed: dnf-0:4.19.0-1.fc40.noarch at Mon Feb 19 08:10:37 2024
>   Built    : Fedora Project at Thu Feb  8 16:33:42 2024
> 
>   Installed: rpm-0:4.19.1.1-1.fc40.x86_64 at Mon Feb 19 08:10:37 2024
>   Built    : Fedora Project at Wed Feb  7 15:55:53 2024
> ```

Maybe I removed it...

```
$ yum
bash: yum: command not found
$ rpm -q yum
package yum is not installed
```